### DISCUSSION TITLE
Another supported modbus tcp-rtu gateway

### DISCUSSION THREAD

**Discussion by joriws** [veista/nilan#13] [General]
I am glad that finally there is better Nilan support for home assistant. I've build my own integration and shared it with the world, but it is intended more reading data than programming settings via automation. I don't see that there is much to program in my Nilan except maybe setting fan speed (1-4) for fireplace ignition or usermode for fireplace.

[nilan2homeassistant](https://github.com/joriws/nilan2homeassistant)

Why I wrote, I just wanted to say that another gateway what seems to work out of box is my RPI Zero W with Mbusd gateway software. Sure you could use microcontroller-based gateway, so this is just an option if you have RPI at your hands and want to use Linux/Raspbian.

I just installed your repo via HACS and gave IP and it worked out of the box. RPI0W + MBUSD docs are on my repo. So perhaps on your readme.md you could write MbusD as a supported modbus-gateway alternative.

I did not try programming new values to Nilan EC (VP18C), just reading the values. So maybe being more specific - reading the modbus registers mostly works. However fan speed settings from service menu nor legionella configuration does not work. There might be others too but those got my attention.

**Comment by veista** [comment]
Hi,
Great, I will add that.

As for the problems: Could you provide me the following info: Integration version, modbus version, device software version. Thanks.

**Comment by veista** [comment]
Your device's modbus version (bus version) and firmware (software) version number would really contribute to this project, especially if you were on the latest beta at the time of your writing.

If you could kindly report these. Thank You!
